database2/ffldb/ldb/leveldb.go

```diff
@@ -0,0 +1,64 @@
+package ldb
+
+import (
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/errors"
+)
+
+// LevelDB defines a thin wrapper around leveldb.
+type LevelDB struct {
+	ldb *leveldb.DB
+}
+
+// NewLevelDB opens a leveldb instance defined by the given path.
+func NewLevelDB(path string) (*LevelDB, error) {
+	// Open leveldb. If it doesn't exist, create it.
+	ldb, err := leveldb.OpenFile(path, nil)
+
+	// If the database is corrupted, attempt to recover.
+	if _, corrupted := err.(*errors.ErrCorrupted); corrupted {
+		log.Warnf("LevelDB corruption detected for path %s: %s",
+			path, err)
+		var err error
+		ldb, err = leveldb.RecoverFile(path, nil)
+		if err != nil {
+			return nil, err
```

Comment:
Please wrap any error coming out of the system with our own message.

Reply:
Created a ticket.